ipatests/test_integration/test_sso.py

```diff
@@ -53,6 +54,40 @@ def keycloak_login(host, username, password, username_fl=None):
         host.run_command(["rm", "-f", "/tmp/keycloak_login.py"])
 
 
+def keycloak_add_user(host, kcadm_pass, username, password=None,
+                      username_fl=None):
+    if username_fl is None:
+        username_fl = username
+    domain = host.domain.name
+    kcadmin_sh = "/opt/keycloak/bin/kcadm.sh"
+    kcadmin = [kcadmin_sh, "config", "credentials", "--server",
+               f"https://{host.hostname}:8443/auth/",
+               "--realm", "master", "--user", "admin",
+               "--password", kcadm_pass]
+
+    tasks.run_repeatedly(host, kcadmin, timeout=60)
+    host.run_command([kcadmin_sh, "create", "users", "-r", "master",
+                      "-s", f"username={username}",
+                      "-s", "enabled=true",
+                      "-s", f"email={username}@{domain}"])
+
+    if password is not None:
+        host.run_command([kcadmin_sh, "set-password", "-r", "master",
+                          "--username", "testuser1", "--new-password",
+                          password])
+
+
+def ipa_set_user_password(host, username, password):
+    temppass = "redhat\nredhat"
+    sendpass = f"redhat\n{password}\n{password}"
+    tasks.kdestroy_all(host)
+    tasks.kinit_admin(host)
+    host.run_command(["ipa", "passwd", username],stdin_text=temppass)
+    host.run_command(["kinit", username], stdin_text=sendpass)
+    tasks.kdestroy_all(host)
+    tasks.kinit_admin(host)
```

Comment:
Similar functions also exist in test_pwpolicy.py and test_commands.py. It seems like a good time to centralize changing and resetting a password, perhaps in tasks.py.

Reply:
I was wondering about that and if I should instead move to tasks.py.  Is there any general naming guide for functions there I should adhere to?

Reply:
I like the two methods in test_pwpolicy, kinit_as_user() and reset_password(), but I'm not exactly partial.

I don't recall any formal naming scheme, we all struggle with it. Do your best and we'll bike-shed it in the PR. The names in the policy test don't completely convey what is going on but maybe start there?